flang/lib/Frontend/FrontendActions.cpp

```diff
@@ -66,6 +69,16 @@ bool PrescanAndSemaDebugAction::BeginSourceFileAction() {
 }
 
 bool CodeGenAction::BeginSourceFileAction() {
+  llvmCtx_ = std::make_unique<llvm::LLVMContext>();
+  if (this->currentInput().kind().GetLanguage() == Language::LLVM_IR) {
+      // Parse the bitcode...
+      SMDiagnostic err;
+      llvmModule_ =  parseIRFile(currentInput().file(), err, *llvmCtx_);
+      if (!llvmModule_)
+        return false;
+    return true;
```

Comment:
The indentation looks off here. `clang-format`?

Reply:
Also, why not just `return {llvmModule_};`?

Reply:
`BeginSourceFileAction` is a [virtual method](https://github.com/llvm/llvm-project/blob/a1dc6d4b83a38a52daab08ad52b68a7b80103709/flang/include/flang/Frontend/FrontendAction.h#L42-L46), which returns a `bool`. Also, `llvmModule_` is only really relevant for code-gen actions, but not so much in other cases (e.g. `-fsyntax-only`).

Reply:
It always looks strange to me to read code like:
```
if (expression)
  return true;
/*else*/
  return false;
```
when
```
return bool{expression};
```
suffices.

Reply:
Sorry, I misread your comment then. Now I see what you meant.

How about `return (nullptr != llvmModule_);`? This would make the intent clearer without sacrificing the quality of the generated code.

Reply:
By the way, I wrote a little test to check code quality.

```c++
int *p;

bool test1() {
  return p;
}

bool test2() {
  return p != nullptr;
}

bool test3() {
  return bool{p};
}
```

I ran `g++ -S -std=c++17 test.cpp` and got 3 identical functions.

```asm
_Z5test1v:
        pushq   %rbp
        movq    %rsp, %rbp
        movq    p(%rip), %rax
        testq   %rax, %rax
        setne   %al
        popq    %rbp
        ret
_Z5test2v:
        pushq   %rbp
        movq    %rsp, %rbp
        movq    p(%rip), %rax
        testq   %rax, %rax
        setne   %al
        popq    %rbp
        ret
_Z5test3v:
        pushq   %rbp
        movq    %rsp, %rbp
        movq    p(%rip), %rax
        testq   %rax, %rax
        setne   %al
        popq    %rbp
        ret
```

Reply:
Haha, I've also experimented: https://godbolt.org/z/MhG6fx6Kv :)